lib/Service/ImportService.php

```diff
@@ -0,0 +1,176 @@
+		$this->createUnknownColumns = $createMissingColumns;
+
+		/** @var IConfig $config */
+		$config = Server::get(IConfig::class);
+
+		try {
+			$storageHome = $this->rootFolder->getUserFolder($this->userId);
+			if ($storageHome->nodeExists($path)) {
+				// TODO is this working if a object storage is used?
+				$completePath = $storageHome->get($path)->getPath();
+				$basePath = $config->getSystemValue('datadirectory');
+				$spreadsheet = IOFactory::load($basePath.$completePath);
+				$this->loop($spreadsheet->getActiveSheet());
+			}
+
+		} catch (NotFoundException|NotPermittedException|NoUserException|InternalError|PermissionError $e) {
+			throw new NotFoundError('path could not be found');
+		}
+
+		return [
+			'found columns' => count($this->columns),
+			'created columns' => $this->countCreatedColumns,
+			'inserted rows' => $this->countInsertedRows,
+			'errors (see logs)' => $this->countErrors,
+		];
+	}
+
+	/**
+	 * @throws PermissionError
+	 * @throws InternalError
+	 */
+	private function loop(Worksheet $worksheet): void {
+		$firstRow = true;
+		foreach ($worksheet->getRowIterator() as $row) {
+			if ($firstRow) {
+				$this->getColumns($row);
+				$firstRow = false;
+			} else {
+				// parse row data
+				$this->createRow($row);
+			}
+		}
+	}
+
+	/*
+	 * @return stringify value
+	 */
+	private function parseValueByColumnType(string $value, Column $column): string {
+		try {
+			$businessClassName = 'OCA\Tables\Service\ColumnTypes\\';
+			/** @noinspection PhpUndefinedMethodInspection */
+			$businessClassName .= ucfirst($column->getType()).ucfirst($column->getSubtype()).'Business';
+			/** @var TextLineBusiness $columnBusiness */
+			$columnBusiness = Server::get($businessClassName);
+			return $columnBusiness->parseValue($value, $column);
+		} catch (NotFoundExceptionInterface|ContainerExceptionInterface $e) {
+			$this->logger->debug('Column type business class not found');
+		}
+		return '';
+	}
+
+	private function createRow(Row $row): void {
+		$cellIterator = $row->getCellIterator();
+		$cellIterator->setIterateOnlyExistingCells(false);
+
+		$i = -1;
+		$data = [];
+		foreach ($cellIterator as $cell) {
+			$i++;
+
+			// only add the dataset if column is known
+			if($this->columns[$i] === '' || !isset($this->columns[$i])) {
+				$this->logger->debug('Column unknown while fetching rows data for importing.');
+				$this->countErrors++;
+				continue;
+			}
+
+			// if cell is empty
+			if(!$cell || $cell->getValue() === null) {
+				$this->logger->info('Cell is empty while fetching rows data for importing.');
+				continue;
+			}
+
+			$data[] = [
+				'columnId' => $this->columns[$i]->getId(),
+				'value' => json_decode($this->parseValueByColumnType($cell->getValue(), $this->columns[$i])),
+			];
+		}
+		try {
+			$this->rowService->createComplete($this->tableId, $data);
+			$this->countInsertedRows++;
+		} catch (PermissionError|Exception $e) {
+			$this->logger->error('Could not create row while importing.');
```

Comment:
Would it make sense to be a bit more verbose about which row (number) failed?

Especially when catching generic exceptions it should most of the time also pass the exception along with the log so that the trace is written.